Polars 23.1 Compile Error with "is_in" feature

#### What language are you using?

Rust

#### Which feature gates did you use?

"is_in"

#### Have you tried latest version of polars?

- [yes]

#### What version of polars are you using?

23.1

#### What operating system are you using polars on?

Windows

#### What language version are you using
ex: python 3.8, node 16, etc.. 

#### Describe your bug.

Polars 23.1 compilation fails whith feature "is_in"

#### What are the steps to reproduce the behavior?
Cargo.toml
polars = {version="0.23.1", features=["is_in"]}
#### Error message:
```
 Compiling polars-lazy v0.23.1
error[E0599]: no variant or associated item named `Categorical` found for enum `polars_core::datatypes::DataType` in the current scope
   --> C:\Users\Anato\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-lazy-0.23.1\src\logical_plan\optimizer\type_coercion.rs:372:32
    |
372 |                     (DataType::Categorical(_), DataType::Utf8) => {
    |                                ^^^^^^^^^^^ variant or associated item not found in `polars_core::datatypes::DataType`

error[E0599]: no variant or associated item named `Categorical` found for enum `polars_core::datatypes::DataType` in the current scope
   --> C:\Users\Anato\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-lazy-0.23.1\src\logical_plan\optimizer\type_coercion.rs:377:50
    |
377 | ...                   data_type: DataType::Categorical(None),
    |                                            ^^^^^^^^^^^ variant or associated item not found in `polars_core::datatypes::DataType`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `polars-lazy` due to 2 previous errors
```

--- a/polars/polars-lazy/src/logical_plan/optimizer/type_coercion.rs
+++ b/polars/polars-lazy/src/logical_plan/optimizer/type_coercion.rs
@@ -369,6 +369,7 @@ impl OptimizationRule for TypeCoercionRule {
                 match (&type_left, type_other) {
                     // cast both local and global string cache
                     // note that there might not yet be a rev
+                    #[cfg(feature = "dtype-categorical")]
                     (DataType::Categorical(_), DataType::Utf8) => {
                         let mut input = input.clone();
 

